hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/QueryModule.java

```diff
@@ -19,25 +19,68 @@
+import com.hedera.node.app.service.contract.impl.exec.scope.SystemContractOperations;
+import com.hedera.node.app.service.contract.impl.exec.utils.ActionStack;
+import com.hedera.node.app.service.contract.impl.hevm.ActionSidecarContentTracer;
+import com.hedera.node.app.service.contract.impl.hevm.HederaEvmBlocks;
+import com.hedera.node.app.service.contract.impl.hevm.HederaEvmContext;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import com.hedera.node.app.service.contract.impl.hevm.QueryContextHevmBlocks;
 import com.hedera.node.app.service.contract.impl.state.EvmFrameStateFactory;
+import com.hedera.node.app.service.contract.impl.state.ProxyWorldUpdater;
 import com.hedera.node.app.service.contract.impl.state.ScopedEvmFrameStateFactory;
+import com.hedera.node.app.service.token.ReadableAccountStore;
 import com.hedera.node.app.spi.workflows.QueryContext;
-import com.swirlds.config.api.Configuration;
 import dagger.Binds;
 import dagger.Module;
 import dagger.Provides;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.function.Supplier;
 
-@Module
+@Module(includes = {QueryConfigModule.class})
 public interface QueryModule {
     @Provides
     @QueryScope
-    static Configuration provideConfiguration(@NonNull final QueryContext context) {
-        return requireNonNull(context).configuration();
+    static ProxyWorldUpdater provideProxyWorldUpdater(
+            @NonNull final HederaOperations extWorldScope,
+            @NonNull SystemContractOperations systemContractOperations,
+            @NonNull final EvmFrameStateFactory factory) {
+        return new ProxyWorldUpdater(
+                requireNonNull(extWorldScope), requireNonNull(systemContractOperations), requireNonNull(factory), null);
+    }
+
+    @Provides
+    @QueryScope
+    static ActionSidecarContentTracer provideActionSidecarContentTracer() {
+        return new EvmActionTracer(new ActionStack());
+    }
+
+    @Provides
+    @QueryScope
+    static Supplier<HederaWorldUpdater> provideFeesOnlyUpdater(
+            @NonNull final HederaOperations extWorldScope,
+            @NonNull SystemContractOperations systemContractOperations,
+            @NonNull final EvmFrameStateFactory factory) {
+        return () -> new ProxyWorldUpdater(
+                requireNonNull(extWorldScope), requireNonNull(systemContractOperations), requireNonNull(factory), null);
+    }
+
+    @Provides
+    @QueryScope
+    static HederaEvmContext provideHederaEvmContext(
+            @NonNull final HederaOperations extWorldScope, @NonNull final HederaEvmBlocks hederaEvmBlocks) {
+        return new HederaEvmContext(extWorldScope.gasPriceInTinybars(), true, hederaEvmBlocks);
+    }
+
+    @Provides
+    @QueryScope
+    static ReadableAccountStore provideReadableAccountStore(@NonNull final QueryContext context) {
+        return requireNonNull(context).createStore(ReadableAccountStore.class);
     }
```

Comment:
Similar to configurations, the pattern is to retrieve stores inside the method where said store is used. No need for dagger here. Not sure, do any of the other objects provided here fall into the same category?